frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -54,6 +54,43 @@ import {
 import './ocs-storage-class-form.scss';
 import { GUARDED_FEATURES } from '../../features';
 
+export const CephFsNameComponent: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+
+  const [cephClusters, loaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+
+  const onParamChangeRef = React.useRef(onParamChange);
+  React.useEffect(() => {
+    if (cephClusters[0]?.metadata.name === CEPH_INTERNAL_CR_NAME) {
+      onParamChangeRef.current('ocs-storagecluster-cephfilesystem');
+    }
```

Comment:
Now when we are showing a custom component, its makes sense to show a disabled input in case of internal mode with no option to edit default value.